### DISCUSSION TITLE
Feedback endpoint /devices dan /messages

### DISCUSSION THREAD

**Discussion by rioastamal** [kirimwa-id/forum#1] [Saran dan Ide]
> Ini adalah feedback dari Tegar Imansyah yang dikirim via email ke Saya (Rio Astamal). Saya salin disini agar member lain bisa mengikuti.

Halo Mas Rio

Aku sudah beberapa kali percobaan dengan API kirimwa.id [1], it works
like a charm. Kondisi percobaanku:

* Request menggunakan postman
* WA menggunakan dual apps (menjalankan 2 wa sekaligus, dimana nomor
untuk kirimwa.id [1] adalah secondary).
   -> Walaupun gak buka app nya (terkesan sleep dan gak ada notif
"whatsapp web is currently active"), tapi ternyata masih aktif dan
tetap bisa mengirim pesan

Next step aku mau mencoba integrasi dengan prototype app yang sedang
aku bikin. Disini nanti harapannya bisa dilihat bagaimana jika
penggunaan lebih aktif. Semoga diizinkan dan diusahakan tetap dibawah
limitasi 100 pesan / hari.

Ohh iya sedikit feedback dari saya:
* untuk /devices , ketika pernah disconnect lalu connect lagi,
_disconnected_at_ dan _disconnected_reason_ tetap muncul.
* untuk endpoint /messages , mungkin bisa ditambahkan "_updated_at_"
agar bisa dihitung berapa lama pesan terkirim semenjak "created_at"
atau dari pembuatan request di sisi server kami.
* untuk endpoint /messages , hanya menampilkan pesan sukses saja. Saya
mencoba untuk mengirim ke nomor yang tidak terdaftar dan fail pada
/messages/<msg id> , tapi tidak muncul pada /messages

Nb: Aku bikin postman collection, jika memang dibutuhkan aku mau share
biar orang lain bisa onboarding lebih mudah.

Salam hangat,
Tegar Imansyah

**Comment by rioastamal** [comment]
Halo Mas Tegar,

Terima kasih banyak sudah mencoba dan memberikan feedback. 

Berikut balasan saya tentang feedback yang dikirimkan.

* Field `disconnected_at` dan `disconnected_reason` pada endpoint `/devices` memang sengaja tetap ditampilkan dengan tujuan tetap mengetahui status atau alasan disconnected terakhir.
* Ya, sepertinya `updated_at` ide yang bagus.
* Endpoint `/messages` by default akan menampilkan `status=success`. Jika ingin menampilkan `status` yang `fail` maka harus secara explicit ditambahkan di query string `/messages?status=fail`. Keterangan lengkapnya ada di https://developer.kirimwa.id/api/v1/#get-messages

Thanks,
Rio Astamal

**Reply by tegarimansyah** [reply]
Thank you mas Rio.

Untuk nomor 3, wah sepertinya saya kelewat untuk qs param nya haha. Endpoint itu aku nyoba hanya dari example quick intro di halaman [landing page](https://developer.kirimwa.id/), ternyata ada keterangan lebih lanjut di redoc.

Karena gak ada `?status=all`, maka asumsi kalo mau dapet data semuanya baik yang success, pending atau fail via webhook ya. Dan jika mau cek history harus hit tiga kali ke endpoint tersebut dengan status berbeda ya.

**Reply by rioastamal** [reply]
> Karena gak ada ?status=all, maka asumsi kalo mau dapet data semuanya baik yang success, pending atau fail via webhook ya. Dan jika mau cek history harus hit tiga kali ke endpoint tersebut dengan status berbeda ya.

Betul. Saat ini tidak bisa menampilkan semua status (`success | pending | fail`) dan harus memilih salah satu. Hal ini karena implementasi database yang saya gunakan. Jadi cara terbaik menurut saya untuk saat ini adalah via webhook.
